### PR TITLE
S6- Generate a booking report

### PR DESCRIPTION
**Pull request by Ashton-Jeffrey**
- Created HTTP test for report route API 
- created report with GET API route

### CODE REVIEW

File: app/Http/Controllers/BookingController.php [hunk 1]
```diff
@@ -85,6 +86,27 @@ public function all(Request $request)
...
+
+    public function getReport()
+    {
+        $report = Type::select('types.id', 'types.name')
+            ->withCount('booking')
+            ->withAvg(['booking' => function ($query) {
+                $query->selectRaw('ROUND(DATEDIFF(end, start)) as average_booking_duration')
+                    ->groupBy('type_id');
+            }], 'id')
+            ->get();
```
Comment (ashleycoles) [inline app/Http/Controllers/BookingController.php:98]: This some derren brown shit, nice work!

File: app/Models/Booking.php [hunk 2]
```diff
@@ -16,4 +17,9 @@ public function room(): BelongsTo
     {
         return $this->belongsTo(Room::class);
     }
+
+    public function type(): HasOneThrough
```
Comment (ashleycoles) [inline app/Models/Booking.php:21]: Is this still being used?

File: app/Models/Type.php [hunk 3]
```diff
@@ -16,4 +17,9 @@ public function rooms(): HasMany
     {
         return $this->hasMany(Room::class);
     }
+
+    public function booking(): HasManyThrough
```
Comment (ashleycoles) [inline app/Models/Type.php:21]: Still needed? (Genuine question)

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: app/Models/Booking.php (hunk 2 above)
Comment (Ashton-Jeffrey) [reply app/Models/Booking.php:21]: its is indeed

File: app/Models/Type.php (hunk 3 above)
Comment (Ashton-Jeffrey) [reply app/Models/Type.php:21]: its is indeed

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**